components/window_size_observer/WindowSizeObserver.tsx

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect} from 'react';
+
+import {useDispatch} from 'react-redux';
+
+import throttle from 'lodash/throttle';
+
+import {setLhsSize} from 'actions/views/lhs';
+import {setRhsSize} from 'actions/views/rhs';
+
+import Constants, {SidebarSize} from 'utils/constants';
+
+const smallSidebarMediaQuery = window.matchMedia(`(max-width: ${Constants.SMALL_SIDEBAR_BREAKPOINT}px)`);
+const mediumSidebarMediaQuery = window.matchMedia(`(min-width: ${Constants.SMALL_SIDEBAR_BREAKPOINT + 1}px) and (max-width: ${Constants.MEDIUM_SIDEBAR_BREAKPOINT}px)`);
+const largeSidebarMediaQuery = window.matchMedia(`(min-width: ${Constants.MEDIUM_SIDEBAR_BREAKPOINT + 1}px) and (max-width: ${Constants.LARGE_SIDEBAR_BREAKPOINT}px)`);
+const xLargeSidebarMediaQuery = window.matchMedia(`(min-width: ${Constants.LARGE_SIDEBAR_BREAKPOINT + 1}px)`);
+
+function WindowSizeObserver() {
```

Comment:
There should as little as possible JS running when resizing the window.. Could all of this logic be included in or launched from here? https://github.com/mattermost/mattermost-webapp/blob/d0230dc4dde704d69874f4d9e795554c8b824eec/actions/views/browser.ts#L8


This seems extraneous. Could the existing breakpoints based on `views.browser.windowSize` be used for some of this?


<img width="400" alt="image" src="https://user-images.githubusercontent.com/11724372/214870673-68b90b36-b0ca-40e4-a71b-a8f87771eec2.png">

Reply:
I thought I should distinguish between `browser.windowSize` and `lhs.size` because breakpoint is applied differently.
To be included in one, I think we need to add more break points to the function `emitBrowserWindowResized`. That would make the function more complicated, would that be okay?

<img width="318" alt="스크린샷 2023-01-31 오전 12 17 58" src="https://user-images.githubusercontent.com/59413880/215517352-d73a5d46-adbc-4af5-8db4-bd273a57d813.png">